arcade/examples/sprite_health.py

```diff
@@ -1,230 +1,249 @@
+        self.health: int = PLAYER_HEALTH
+
+
+class IndicatorBar:
+    """
+    Represents a bar which can display information about a sprite.
+
+    Parameters
+    ----------
+    owner: Player
+        The owner of this indicator bar.
+    sprite_list: arcade.SpriteList
+        The sprite list used to draw the indicator bar components.
+    position: tuple[float, float]
+        The initial position of the bar.
+    full_color: arcade.Color
+        The color of the bar.
+    background_color: arcade.Color
+        The background color of the bar.
+    width: int
+        The width of the bar.
+    height: int
+        The height of the bar.
+    border_size: int
+        The size of the bar's border.
+    """
+
+    def __init__(
+        self,
+        owner: Player,
+        sprite_list: arcade.SpriteList,
+        position: tuple[float, float] = (0, 0),
+        full_color: arcade.Color = arcade.color.GREEN,
+        background_color: arcade.Color = arcade.color.BLACK,
+        width: int = 100,
+        height: int = 4,
+        border_size: int = 4,
+    ) -> None:
+        # Store the reference to the owner and the sprite list
+        self.owner: Player = owner
+        self.sprite_list: arcade.SpriteList = sprite_list
+
+        # Set the needed size variables
+        self._box_width: int = width
+        self._box_height: int = height
+        self._half_box_width: int = self._box_width // 2
+        self._center_x: float = 0.0
+        self._center_y: float = 0.0
+        self._fullness: float = 0.0
+
+        # Create the boxes needed to represent the indicator bar
+        self._background_box: arcade.SpriteSolidColor = arcade.SpriteSolidColor(
+            self._box_width + border_size,
+            self._box_height + border_size,
+            background_color,
+        )
+        self._full_box: arcade.SpriteSolidColor = arcade.SpriteSolidColor(
+            self._box_width,
+            self._box_height,
+            full_color,
+        )
+        self.sprite_list.append(self._background_box)
+        self.sprite_list.append(self._full_box)
+
+        # Set the fullness and position of the bar
+        self.fullness: float = 1.0
+        self.position: tuple[float, float] = position
+
+    def __repr__(self) -> str:
+        return f"<IndicatorBar (Owner={self.owner})>"
+
+    @property
+    def background_box(self) -> arcade.SpriteSolidColor:
+        """Returns the background box of the indicator bar."""
+        return self._background_box
+
+    @property
+    def full_box(self) -> arcade.SpriteSolidColor:
+        """Returns the full box of the indicator bar."""
+        return self._full_box
+
+    @property
+    def fullness(self) -> float:
+        """Returns the fullness of the bar."""
+        return self._fullness
+
+    @fullness.setter
+    def fullness(self, new_fullness: float) -> None:
+        """Sets the fullness of the bar."""
+        # Check if new_fullness if valid
```

Comment:
I would add a blank line between the docstring and the comment to increase legibility.